integrated_channels/sap_success_factors/client.py

```diff
@@ -373,6 +373,228 @@ def _call_post_with_session(self, url, payload):
             )
         return response.status_code, response.text
 
+    def get_user_details(self, logged_in_user_id):
```

Comment:
This method is too long and can be broken down into smaller private methods. Maybe we can create a different method for each step?